subsys/net/ip/net_mgmt.c

```diff
@@ -240,12 +240,14 @@ static int mgmt_event_wait_call(struct net_if *iface,
 	net_mgmt_add_event_callback(&sync);
 
 	ret = k_sem_take(sync.sync_call, timeout);
+
+	net_mgmt_del_event_callback(&sync);
```

Comment:
Hmm, the thing is, the mgmt thread removes this event callback before feeding the semaphore (https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/ip/net_mgmt.c#L175). So in case `k_sem_take()` reports no error, the callback shouldn't even be present on the list anymore.

Reply:
Ah you're absolutely right - internally I was using an older release that was missing #62302. Incorporating that and will close this PR.